W3C.Domain/Repositories/MongoDbRepositoryBase.cs

```diff
@@ -62,12 +80,18 @@ protected IMongoCollection<T> CreateCollection<T>(string collectionName = null)
 
     protected async Task Upsert<T>(T insertObject, Expression<Func<T, bool>> identityQuerry)
     {
-        var mongoDatabase = CreateClient();
-        var mongoCollection = mongoDatabase.GetCollection<T>(typeof(T).Name);
-        await mongoCollection.FindOneAndReplaceAsync(
-            identityQuerry,
-            insertObject,
-            new FindOneAndReplaceOptions<T> { IsUpsert = true });
+        var mongoCollection = CreateCollection<T>();
+        var session = _transactionCoordinator?.GetCurrentSession();
+        var options = new FindOneAndReplaceOptions<T> { IsUpsert = true };
+
+        if (session != null)
```

Comment:
Maybe good to be consistent with using ternary or not. 40 lines up a ternary is used for this exact same thing